docs/high_availability_installation_guide.md

```diff
@@ -0,0 +1,284 @@
+
+### Stateless service 
+To make a stateless service highly available, you need to provide redundant instances and load balance them. Harbor services that are stateless include: 
+ - Adminserver
+ - UI
+ - Registry
+ - Logs
+ - Jobservice
+ - Clair
+ - Proxy 
+
+### Stateful service 
+
+Stateful services are more difficult to manage. Providing additional instances and load balancing does not solve the problem. Harbor services that are stateful include the follow services:
+ - Harbor database(MariaDB)
+ - Clair database(PostgresSQL)
+ - Notary database(MariaDB)
+ - Redis
+Making those services highly available can depend the high available implementation for each application. 
+
+## High Availability Architecture 
+
+Again, this architecture is for advice only. 
+![HA](img/ha/Architecture.png)
+As seen in the figure, components involved in the architecture are: 
+ 
+**VIP**: Virtual IP. The Harbor user will access Harbor through this virtual IP address. This VIP will only active on one load balancer node at the same time. It will automatically switch to the other node if the active loadbalancer node is down. 
+
+**LoadBalancer01 and 02**: They together compose as a group which avoid single point failure of load balancer nodes. Keepalived is installed on both load balancer nodes. The two Keepalived instance will form a VRRP group to provide the VIP and ensure the VIP only shows on one node at the same time. The LVS component in Keepalived is responsible for balance the requests between different Harbor servers according to the routing algorithm.  
+
+**Harbor server 1..n**: These are the running Harbor instances. They are in active-active mode. User can setup multiple nodes according to their workload. 
+
+**Harbor DB cluster**: The MariaDB is used by Harbor to store user authentication information, image metadata information and so on. User should follow its best practice to make it HA protected. 
+
+**Clair DB cluster**: The PostgreSQL is used by Clair to store the vulnerability data which will be used by scanning the images. User should follow it's best practice to make it HA protected 
+
+**Shared Storage**: The shared storage is used for storing Docker Volumes used by Harbor. Images pushed by users are actually stored in this shared storage. The shared storage makes sure that multiple Harbor instances have consistent storage backend. Shared Storages can be Swift, NFS, S3, azure, GCS Ceph or OSS. User should follow its best practice to make it HA protect. 
+
+**Redis**: The purpose of having Redis is to store UI session data and store the registry metadata cache. When one Harbor instance fails or the load balancer routes a user request to another Harbor instance, any Harbor instance can query the Redis to retrieve session information to make sure the end-user has a continued session. User should follow the best practice of Redis to make it HA protect.  
```

Comment:
how about: continued session => sticky session

Reply:
Continued session have different mean with session sticky.

Reply:
sorry used the wrong term 'sticky session'..